### PR TITLE
Finesse teams channels.

### PR DESCRIPTION
**Pull request by adamjspooner**
@keybase/y2ksquad, this is ready for review.

**In this PR:**
- Section dividers now use blueGrey for their background
- Team channel helper text resizes (with a maximum width)
- Team channel header's team name underlines when hovered
- Editing a team's channel uses the new team channel edit modal

cc @keybase/react-hackers

### CODE REVIEW

File: shared/teams/channel/header.tsx [hunk 1]
```diff
@@ -148,8 +157,8 @@ const HeaderTitle = (props: HeaderTitleProps) => {
   )
 
   const tip = (
-    <Kb.Box2 direction="horizontal" alignSelf="flex-start" gap="tiny">
-      <Kb.Icon color={Styles.globalColors.black_20} fontSize={12} type="iconfont-info" />
+    <Kb.Box2 direction="horizontal" alignSelf="flex-start" gap="tiny" style={styles.tipBox}>
+      <Kb.Icon color={Styles.globalColors.black_20} type="iconfont-info" sizeType="Small" />
       <Kb.Text type="BodySmall" lineClamp={3}>
```
Comment (buoyad) [inline shared/teams/channel/header.tsx:162]: On narrower window widths this text ellipsizes. Should it?

File: shared/teams/team/rows/channel-row/channel.tsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ const ChannelRow = (props: ChannelRowProps) => {
   const activityLevel = Container.useSelector(
     state => state.teams.activityLevels.channels.get(channel.conversationIDKey) || 'none'
   )
+  const channelMeta = useChannelMeta(teamID, conversationIDKey)
```
Comment (buoyad) [inline shared/teams/team/rows/channel-row/channel.tsx:33]: shouldn't need this, channelname and description are on the `channel` obj above

Comment (jakob223) [reply shared/teams/team/rows/channel-row/channel.tsx:33]: :o sorry I missed this

File: shared/teams/channel/header.tsx (hunk 1 above)
Comment (adamjspooner) [reply shared/teams/channel/header.tsx:162]: It never gets that small, but I'll drop the clamp. 👍

**Pull request merged**